statistics/pooled_prevalence.R

```diff
@@ -0,0 +1,100 @@
+library(tidyverse)
+library(rstan)
+
+options(mc.cores = parallel::detectCores())
+# rstan_options(auto_write = TRUE)
+
+stan_model <- stan_model(model_code = "
+data {
+  // Pooled testing data
+  int<lower=0> N_pools;              // number of pools
+  int<lower=0> x[N_pools];           // test results (0 = negative, 1 = positive)
+  int<lower=1> n[N_pools];           // size of each pool
+  
+  // MGS data
+  int<lower=0> N_mgs;                // number of MGS samples
+  int<lower=0> viral_reads[N_mgs];   // observed viral reads in each MGS sample
+  int<lower=0> total_reads[N_mgs];   // total reads in each MGS sample
+  
+  // Probability a positive individual makes a pool positive
+  real<lower=0, upper=1> sensitivity;
+}
+
+parameters {
+  real<lower=0, upper=1> p;          // prevalence (fraction of positive individuals)
+  real b;                            // log10(p2ra factor)
+  real<lower=0> phi;                 // dispersion parameter for negative binomial
+}
+
+model {
+  // Priors
+  p ~ beta(1, 1);                    // Uniform prior on prevalence
+  b ~ normal(-6, 4);                 // Derived from eyeballing P2RA report and picking a wide range
+  phi ~ exponential(1);              // Prior for dispersion parameter
```

Comment:
This seems like a lot of dispersion to me. Here's my logic (disclaimers like "my first time reading Stan code" apply).

The negative binomial 2 parameterization with parameters `mu, phi` has [variance](https://mc-stan.org/docs/functions-reference/unbounded_discrete_distributions.html#nbalt) `mu + mu^2 / phi`. Given that a `Poisson(mu)` has variance `mu`, this means we're increasing the variance by a factor of `1 + mu/phi`.

For some pathogens, we might expect to see `mu` in the hundreds, right? In which case if `phi` is around 1 -- not crazy for `phi ~ exponential(1)` -- we'd be saying the viral counts have variance hundreds of times higher than just Poisson counting noise? Seems like a lot?

(I guess this is a general problem of the `mu^2` in the variance.)